File: algorithmProblemSolving/programmers/level1/정수 제곱근 판별.cpp
#include <string>
#include <vector>

using namespace std;

long long solution(long long n) {
    long long answer = 0;
    long long tmp;
    for(long long i = 0;i<=n;i++){
        tmp = i*i;
        if(tmp==n){
            return (i+1)*(i+1);
        }else if(tmp>n){
            return -1;
        }
    }
    return answer;
}

File: algorithmProblemSolving/programmers/level1/정수 제곱근 판별.js
/*function solution(n) {
    return Math.pow(parseInt(Math.sqrt(n)),2)===n?Math.pow((Math.sqrt(n)+1),2):-1;
}*/
function solution(n){
    var result = 0;
    var x = 0;
    while (x*x < n){
        x++;
    }
    if (x*x == n){
        x++;
        result = x*x; 
    }else{
        result = -1;
    }
    return result;
}